.github/actions/build-and-push/action.yaml

```diff
@@ -0,0 +1,54 @@
+name: Build and Push
+description: Build and Push docker image to dibbs-query-connector registry
+
+inputs:
+  username:
+    description: GitHub username of the user who triggered the workflow
+    required: true
+  password:
+    description: Used to log in to the Container registry
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Check Out Changes
+      uses: actions/checkout@v4
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: Log in to the Container registry
+      uses: docker/login-action@v2
+      with:
+        registry: ghcr.io
+        username: ${{ inputs.username }}
+        password: ${{ inputs.password }}
+
+    - name: lowercase repo name to use in following step
+      shell: bash
+      env:
+        REPO: ${{ github.repository }}
+      run: |
+        echo "REPO=${GITHUB_REPOSITORY,,}" >> $GITHUB_ENV
+
+    - name: Build and push
+      id: build_push
+      uses: docker/build-push-action@v3
+      with:
+        context: ./query-connector
+        push: true
+        tags: |
+          ghcr.io/${{ env.REPO }}/query-connector:main, ghcr.io/${{ env.REPO }}/query-connector:latest
```

Comment:
We're going to need some sort of hash or semantic versioning to be added here. `main` and `latest` will not reliably trigger cloud platforms to pull and deploy a new image if one already exists with these tags.

Reply:
In addition, does `env.REPO` contain the `CDCgov` part of your tag? Again, I recommend simplifying this.

Reply:
For triggering the new image to be pulled and deployed, the up_helm_chart job triggers the workflow in phdi-charts to create a [PR](https://github.com/CDCgov/phdi-charts/pull/63) that increases the chart version. I linked the PR that adds this workflow in phdi-charts.

Reply:
I modified the [tag](https://github.com/CDCgov/dibbs-query-connector/blob/5a59606aac905d13e61a20d04e533ddd9180dfc7/.github/actions/build-and-push/action.yaml#L40) and the path in a [PR](https://github.com/CDCgov/phdi-charts/pull/64) for phdi-charts